### ISSUE TITLE
Visual Studio naming conventions ignores when dotnet_naming_symbols.<symbol_name>.required_modifiers style is defined for a type

### ISSUE THREAD

**Issue by TsengSR** [dotnet/roslyn#22094]
**Version Used**:
Visual Studio 15.3.4 with Microsoft Code Analyzer (2.3.0.62003) and C# 7.1 compiler

**Steps to Reproduce**:

I tried to set up the following naming convention for a solution:

```
## interface prefix
dotnet_naming_rule.any_interface_prefix.symbols  = any_interface
dotnet_naming_rule.any_interface_prefix.style    = starts_with_i
dotnet_naming_rule.any_interface_prefix.severity = error

dotnet_naming_symbols.any_interface.applicable_kinds           = interface
dotnet_naming_symbols.any_interface.applicable_accessibilities = *
dotnet_naming_symbols.any_interface.required_modifiers = * # this here causes the issue

dotnet_naming_style.starts_with_i.required_prefix = I
dotnet_naming_style.starts_with_i.capitalization  = pascal_case

dotnet_naming_style.have_pascal_casing.capitalization  = pascal_case
```
Naively I used `required_modifiers = *` to take any value. But the rule did not work and no error was shown. After trying around and searching for a solution, I came to the conclusion that this line is causing the issue. 

When I remove the 
```
dotnet_naming_symbols.any_interface.required_modifiers = * # this here causes the issue
```

line, everything works. 

**Expected Behavior**:
That either the `dotnet_naming_symbols.any_interface.required_modifiers = *` is ignored when not applicable for a given `dotnet_naming_symbols.<symbol_name>.applicable_kinds` setting or display an error in some way to tell the user to remove the `required_modifiers`

**Actual Behavior**:
No error message and rule is not applied and it's unclear what's causing the issue.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback on this issue.
